src/gg.regression.unity.bots/Editor/Scripts/RGSettingsUIRegistrar.cs

```diff
@@ -62,29 +67,14 @@ public static SettingsProvider CreateRGSettingsProvider()
                     featureStateRecordingAndReplay.boolValue =
                         EditorGUILayout.Toggle("State Recording & Replay", featureStateRecordingAndReplay.boolValue);
 
-                    if ((EditorApplication.timeSinceStartup - timeOfLastEdit) > 3f && token == null && priorPassword == null && priorHost == null && priorUser == null && passwordField.stringValue.Length > 4 && emailField.stringValue.Length > 4 && hostField.stringValue.Length > 4)
-                    {
-                        priorPassword = passwordField.stringValue;
-                        priorUser = emailField.stringValue;
-                        priorHost = hostField.stringValue;
-                        await Login(priorUser, priorPassword);
-                    }
-
                     settings.ApplyModifiedProperties();
-                    userSettings.ApplyModifiedProperties();
                     if (EditorGUI.EndChangeCheck())
                     {
-                        timeOfLastEdit = EditorApplication.timeSinceStartup;
-                        if (priorHost != hostField.stringValue || priorUser != emailField.stringValue || priorPassword != passwordField.stringValue)
-                        {
-                            priorHost = null;
-                            token = null;
-                            priorPassword = null;
-                            priorUser = null;
-                        }
+                        // This next line is purely for the UI - since we aren't using the apiKeyField.stringValue for
+                        // the UI, we need to update it within the settings. It then still gets saved in the line after.
+                        RGSettings.GetOrCreateSettings().ApiKey = tempApiKey;
```

Comment:
@RG-nAmKcAz this is the fishy part - in order to make sure the UI actually shows the right value, since I am not using the direct property, I need to have a field that is not connected to the serialized property, and then set it later. From my testing, this seems to work entirely correctly, but we no longer will get undo capabilities on the editor field.